packages/client/src/schedule-helpers.ts

```diff
@@ -240,6 +245,7 @@ export async function encodeScheduleAction(
   action: CompiledScheduleAction,
   headers: Headers
 ): Promise<temporal.api.schedule.v1.IScheduleAction> {
+  const jsonConverter = new JsonPayloadConverter();
```

Comment:
Use the data converters configured on the client (i.e. custom payload converter and codec).